src/main/java/org/opensearch/knn/index/engine/MethodComponent.java

```diff
@@ -1,15 +1,9 @@
 /*
+ * Copyright OpenSearch Contributors
  * SPDX-License-Identifier: Apache-2.0
- *
- * The OpenSearch Contributors require contributions made to
- * this file be licensed under the Apache-2.0 license or a
- * compatible open source license.
- *
- * Modifications Copyright OpenSearch Contributors. See
- * GitHub history for details.
  */
 
-package org.opensearch.knn.index;
+package org.opensearch.knn.index.engine;
```

Comment:
Not engine specific, need a better place for this, this holds request parameters for indexing

Reply:
I think with the current hierarchy it is specific to Engine. Engines are composed of KNNMethods wqhich are composed of MethodComponents which are composed of Parameters. I want to move towards encapsulating details around methods behind engines. Hence, I think it makes sense for them to be in the engine package.

Reply:
> I want to move towards encapsulating details around methods behind engines.

I took a deeper look. MethodComponent encapsulates properties of engine in a way and the context is the one which is a request object. So MethodComponent makes sense to be in the engine but MethodComponentContext doesn't because it holds the request given by the user, that isn't related to engine. 

So I would say anything suffixed `Context` should not be in engine as it has parsing logic of user request and then holds the information to create index 

P.S we need naming which doesn't cause confusion

Reply:
I initially wrote the KNNMethodContext to be user configuration of KNNMethod and similarly for MethodComponentContext and MethodComponent. So KNNMethod and MethodComponent define the structure and KNNMethodContext and MethodComponentContext provide the configuration. They have roughly followed this, but not completely. That being said, do you have recommendation for which package they should be in? It seemed to me that for now they could be in the same package because of this parallel.

Reply:
My preference would keep it outside `engine` to start with. `index.request` might be a good place